From 176f10bbe5367336be2f851d8dcce25a569560fd Mon Sep 17 00:00:00 2001
From: Varun Sharma <vsharma@habana.ai>
Date: Mon, 05 Jul 2021 07:42:28 +0300
Subject: [PATCH] [SW-0] gaudi2: use strict-align flag for arm

Add strict-align flag to GCC compilation flags for preboot arm
compilation. This ensures the GCC is aware that the CPU is unable to
handle access to unaligned data.
This will create assembly according to arm64 alignment requirements.
Without this we were getting unaligned exception in gaudi2 preboot.

Change-Id: Ib893664c0be231300fe879be1c4b032d2f31f5eb
Signed-off-by: Varun Sharma <vsharma@habana.ai>
---

diff --git a/arch/gaudi2/Makefile b/arch/gaudi2/Makefile
index cf0367e..2f44d26 100644
--- a/arch/gaudi2/Makefile
+++ b/arch/gaudi2/Makefile
@@ -1,4 +1,5 @@
 FW_SIZE := 0x4400
 PPBOOT_MAX_SIZE := 20480
 
+CFLAGS += -mstrict-align
 image_validation: image_validation_common
